更新FreeModbus主机及从机

armink opened the pull request:
1、更新FreeModbus主机及从机代码至最新版本；
2、移植采用RTT自带的设备驱动框架，降低在RTT平台上的移植难度。

BernardXiong commented:
What's the newest version? We shall remove the version number on the component directory name. Please give your pull request for the rename of directory. Thanks.

armink commented:
The update log which from this old version to newest:
1. added modbus master error process callback method. 
2. changed modbus master request to synchronous method.
3. changed port events to rtt event model.
4. added support modbus master broadcast.
5. fixed some bugs.
